testing/src/main/java/com/linecorp/decaton/testing/TestUtils.java

```diff
@@ -189,20 +200,15 @@ public static void awaitCondition(String message,
      * @param timeoutMillis max duration to wait
      */
     public static void awaitCondition(String message,
-                                      Supplier<Boolean> condition,
+                                      BooleanSupplier condition,
                                       long timeoutMillis) {
         long start = System.nanoTime();
-        while (!condition.get()) {
+        while (!condition.getAsBoolean()) {
             long elapsedMillis = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - start);
             if (elapsedMillis >= timeoutMillis) {
                 throw new AssertionError(message);
             }
-            try {
-                Thread.sleep(100L);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new RuntimeException(e);
-            }
+            LockSupport.parkNanos(TimeUnit.MILLISECONDS.toNanos(100));
```

Comment:
What's the intention of this change? Seems like waiting just 100ms here isn't an actual problem?

Reply:
Just avoid having to deal with `InterruptedException`. No strong preference anyways